cuml/kmeans_demo.ipynb

```diff
@@ -0,0 +1,271 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# KMeans\n",
+    "\n",
+    "KMeans is a basic but powerful clustering method which is optimized via Expectation Maximization. It randomnly selects K data points in X, and computes which samples are close to these points. For every cluster of points, a mean is computed (hence the name), and this becomes the new centroid.\n",
+    "\n",
+    "cuML’s KMeans supports the fast KMeans++ intialization method. This method is more stable than randomnly selecting K points.\n",
+    "\n",
+    "<br>\n",
+    "\n",
+    "|  Parameters: |  Description:    |\n",
+    "| ------------ | ---------------- |\n",
+    "| `n_clusters:int (default = 8)` | The number of centroids or clusters you want. |\n",
+    "| `max_iter:int (default = 300)` | The more iterations of EM, the more accurate, but slower. |\n",
+    "| `tol:float (default = 1e-4)` | Stopping criterion when centroid means do not change much. |\n",
+    "| `verbose:boolean (default = 0)` | If True, prints diagnositc information. |\n",
+    "| `random_state:int (default = 1)` | If you want results to be the same when you restart Python, select a state. |\n",
+    "| `precompute_distances:boolean (default = ‘auto’)` | Not supported yet. |\n",
+    "| `init: (default = 'scalable-k-means++')` | 'scalable-k-means++' or 'k-means\\|\\|': Uses fast and stable scalable kmeans++ intialization. <br> 'random' or an ndarray: Choose 'n_cluster' observations (rows) at random from data for the initial centroids. <br> If an ndarray is passed, it should be of shape (n_clusters, n_features) and gives the initial centers. |\n",
+    "| `n_init:int (default = 1)` | Number of times intialization is run. More is slower, but can be better. |\n",
+    "| `algorithm:“auto”` | Currently uses full EM, but will support others later. |\n",
+    "| `n_gpu:int (default = 1)` | Number of GPUs to use. If -1 is used, uses all GPUs. More usage is faster. |\n",
```

Comment:
Unfortunately, this argument is not even being used in kmeans currently and should be removed. I will open an issue to remove this from the Python code and other docs